### PR TITLE
Remove stream

### PR DESCRIPTION
**Pull request by botronic**
Implements a new method for removing a video frame and the associated stream. Includes moving up all video frames "after" the target frame to remove any empty squares. Addresses #42

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -114,23 +110,79 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
...
 
             self.add_new_stream()
+    
+    def delete_stream(self, videoframe):
+        """Removes selected stream/videoframe from grid"""
+
+        videoframe.hide()
+
+        #If one or two frames
+        if (len(self.videoframes) < 3):
```
Comment (mictab) [inline main.py]: Remove parentheses.

File: main.py [hunk 2]
```diff
@@ -114,23 +110,79 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
...
+
+        videoframe.hide()
+
+        #If one or two frames
+        if (len(self.videoframes) < 3): 
+            #Reset coordinates and delete frame
+            self.coordinates = StreamCoordinates(x=0, y=0)
+            self.delete_videoframe(videoframe)
+            #If one frame left after deletion
+            if (len(self.videoframes) == 1):
```
Comment (mictab) [inline main.py]: Remove parentheses.

File: main.py [hunk 3]
```diff
@@ -114,23 +110,79 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
...
+                #Set last frame's coordinates to (0,0) and update coordinates
+                last_frame = self.videoframes[0]
+                last_frame._coordinates = self.coordinates
```
Comment (mictab) [inline main.py]: Will this work? What's the type of `_coordinates`?

File: main.py [hunk 4]
```diff
@@ -114,23 +110,79 @@ def add_new_stream(self, *args, stream_url=None, stream_quality="480p30"):
...
+                self.update_new_stream_coordinates()
+        #Otherwise there are more frames
+        else:
+            #Get coordinates of frame
+            x = videoframe._coordinates.x
+            y = videoframe._coordinates.y
+            self.coordinates = StreamCoordinates(x=x, y=y)
+
+            print ("(" +  str(x) + ", " + str(y) + ")")
```
Comment (mictab) [inline main.py]: Remove print statements when they're not needed anymore for debugging.

File: main.py (hunk 3 above)
Comment (botronic) [inline main.py]: They are StreamCoordinates. They are added in the setup_stream method.

File: main.py (hunk 4 above)
Comment (botronic) [inline main.py]: Wops 💃

**Comment by mictab** [comment]
Man lever ju ändå bara en gång.

**Comment by botronic** [comment]
<3

**Pull request merged**